### DISCUSSION TITLE
How to install Ruby2 or RVM

### DISCUSSION THREAD

**Discussion by zjmok** [termux/termux-packages#10268] [Q&A] [answered]
I want to install Ruby2.6, but the official script of the RVM cannot run on Termux, I tried to build Ruby2 with Termux-Packages source code, but failed. Is there any way?

**Comment by xtkoba** [comment, marked as answer]
First note that Ruby 2.6 has reached EOL. I never recommend installing it.

That said, surely it was once built successfully in our CI. But not sure if it builds on device. How it fails? Is there any build log that can be posted here?

**Reply by zjmok** [reply]
docker container: 
```
18 warnings generated.
compiling /root/.termux-build/ruby/src/ext/socket/ifaddr.c
installing default socket libraries
30 warnings generated.
linking shared-object socket.so
make[2]: Leaving directory '/root/.termux-build/ruby/build/ext/socket'
make[1]: Leaving directory '/root/.termux-build/ruby/build'
make: *** [uncommon.mk:286: build-ext] Error 2
root@c22de0ae6b01:~/termux-packages#
```
I cloned the latest source code and replaced `packages/ruby` with version 2.6.5 (commit id: 5123b0cffaad65ca19fd18001a1401ed787bf9eb)

**Reply by xtkoba** [reply]
Please include the context of the *first error* occurring in the build. If unsure please share the entire log. Thanks.

**Reply by zjmok** [reply]
This is so esoteric. 

build log:  
[build_ruby_2.6.5.log](https://github.com/termux/termux-packages/files/8549131/build_ruby_2.6.5.log)

error:  
```
make[2]: *** [Makefile:314: ossl_pkey_rsa.o] Error 1
...
make[1]: *** [exts.mk:242: ext/openssl/all] Error 2
...
make: *** [uncommon.mk:286: build-ext] Error 2
```

```
$ openssl version
OpenSSL 1.1.1l  24 Aug 2021
```

Now I don't know what to do 。。。Thank you for your help.

**Reply by zjmok** [reply]
I checkout the source code of ruby2.7.2, replaced NDK in Docker, and replaced the invalid url and sha256. finally the compilation was successful.
